pyrobosim_ros/examples/demo_pddl_goal_publisher.py

```diff
@@ -24,8 +24,13 @@ def __init__(self):
             GoalSpecification, "goal_specification", 10
         )
         self.get_logger().info("Waiting for subscription")
-        while self.goalspec_pub.get_subscription_count() < 1:
+        while rclpy.ok() and self.goalspec_pub.get_subscription_count() < 1:
```

Comment:
Seems like this change causes a deadlock when you run e.g. `ros2 launch pyrobosim_ros demo_pddl.launch.py`. Can you look at this to see if something is amiss, or else revert this?

Reply:
I don't see what the problem is. I figured it might be that rclpy.init() was not called prior, but it is.

I'll refactor and open a new PR to avoid the issue.